lib/features/splash/splash_screen.dart

```diff
@@ -10,34 +15,43 @@ class SplashScreen extends StatefulWidget {
 }
 
 class _SplashScreenState extends State<SplashScreen> {
-  @override
-  void initState() {
-    super.initState();
-    nextAction();
-  }
-
-  Future<void> nextAction() async {
-    await Future.delayed(const Duration(seconds: 3), () {});
-    if (!mounted) return;
-    await Navigator.pushNamedAndRemoveUntil(
-      context,
-      RouteNames.dashboard,
-      (route) => false,
+  void _redirectToPage(
+    BuildContext context,
+    String routeName,
+  ) {
+    WidgetsBinding.instance.addPostFrameCallback(
+      (_) => GoRouter.of(context).goNamed(routeName),
     );
   }
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      body: Container(
-        margin: const EdgeInsets.all(40),
-        decoration: const BoxDecoration(
-          image: DecorationImage(
-            image: AssetImage(AppAssets.imgDroidcon),
-            fit: BoxFit.fitWidth,
+    return ValueListenableBuilder(
+      valueListenable:
+          Hive.box<dynamic>(FlutterConConfig.instance!.values.hiveBox)
+              .listenable(),
+      builder: (context, _, __) {
+        final accessToken = getIt<HiveRepository>().retrieveToken();
+
+        if (accessToken == null) {
```

Comment:
Hmm...for this particular case I'm not a fan of having the redirect logic on the build method (or side effects in general). 

I'd suggest we have similar logic to what was previously present.

i.e. 

initstate() {
  ... 
  final accesstoken = getIt<....;
  ...
  _redirectToPage(...)
}